src/mappers/provinces/province_mapper_importer.cpp

```diff
@@ -44,6 +46,42 @@ void VerifyProvinceIsMapped(const mappers::Hoi4ToVic3ProvinceMapping& hoi4_to_vi
    }
 }
 
+std::vector<std::string> AreVic3ProvincesFromSameState(const std::vector<std::string>& provinces_from_map,
+    const std::map<std::string, vic3::StateRegion>& state_regions_from_save)
+{
+   std::map<std::string, std::string> province_to_state_map;
```

Comment:
It'd be more efficient to generate this once for all provinces and store it as part of the class, rather than regenerating it for each link.